Quick Start Documentation/FPGA Addon Quick Start Guide.md

```diff
@@ -0,0 +1,282 @@
+
+<style>
+    h1{
+        color:#87dfd6;
+    }
+    h2{
+        color:#01a9b4;
+    }
+</style>
+
+
+
+
+# Overview
+
+This add-on runs a FPGA bitfile without needing to implement the normal VeriStand FPGA template. It allows the transfer of basic scalar data types inline with VeriStand's primary control loop (PCL). It also adds support for reading/writing waveforms with FPGA DMA channels.​
+
+<br>
+<br>
+
+# Repo location
+
+[Github](https://github.com/ni/niveristand-fpga-addon-custom-device)
+
+---
+<br>
+
+# Requirements
+
+These drivers must be installed on hte deployment computer/target to function:
+
+- NI VeriStand 2018 or later
+- NI RIO 18.0 or later
+---
+<br>
+
+# FPGA Specifics
+
+To use the addon, first create an FPGA program. Use the provided FPGA IP (discussed below) or custom logic. Put items of interest to front panel controls and indicators. Put waveforms of interest to DMA channels. __It is a requirement to the FPGA code to have an IRQ (address of 30) to control when the FPGA bitfile starts.__ This IRQ only needs to control when DMA read/write operations start. Any other code can start whenever it makes sense. Below is an example showing DMA, controls, and indicators.
+
+
+![Getting Started](./Images/image002.png)
+
+---
+
+
+# Supported Datatypes
+
+## Scalar Datatypes (Controls and indicators)
+
+- Bool
+- U8
+- U16
+- U32
+- U64
+- I8
+- I16
+- I32
+- I64
+- SGL
+- FXP +16.3
+- FXP +/-16.5
+- FXP +/-16.7
+- FXP +18.3
+- FXP +/-20.5
+- FXP +/-24.5
+- FXP +/-27.5
+- FXP +/-32.8
+- FXP +/-32.16
+- FXP +/-64.32
+
+**Note**: Enum​s are not supported. Conve​rt them to a U8, U16, or U32 control or indicator. **Clusters are not supported.** Even if the clusters are made up of supported types. Items can be grouped by naming them on the FPGA with a **_GroupName.SignalName_** format. Support for grouping can only be set on the initialization screen.
+
+---
+<br>
+
+## DMA Datatypes
+
+- U8
+- I8
+- U16
+- I16
+- U32
+- I32
+- U64
+- I64
+- SGL
+- FXP +/-32.8
+- FXP +/-32.16
+- FXP +/-64.32
+
+---
+**Note** The unsigned integer data types can be converted into a bitpacked Boolean array. This is discussed in detail below.
+
+<br>
+
+## Specialty IO
+
+The addon comes with special IP for doing the following:
+
+- PWM Generation - **PWM.Generate.Digital(.vi)**
+- Sinewave Generation - **Sine.Generate(.vi)**
+- Digital Static Generation  - **Static.Generate.Digital(.vi)**
+- Analog PWM Measurement - **PWM.Measure.Analog(.vi)**
+- Digital PWM Measurement - **PWM.Measure.Digital(.vi)**
+- Digital Static Measurement - **Static.Measure.Digital(.vi)**
+- Digital Wheel Speed Sensor Simulation - **WSS.Generate.Digital(.vi)**
+- Quadrature Encoder Measurement - **QuadratureEncoder.Measure.Digital(.vi)**
+
+This IP is located at:
+
+_C:\Users\Public\Documents\National Instruments\NI VeriStand **_Year_**\Custom Devices\FPGA Addon\FPGA IP​_ -> Location is to still be discussed. 
```

Comment:
Regarding these placeholders for locations, didn't we cover them in a previous talk? If yes, then let's go ahead and put the actual locations here and update any user stories with these required paths.

Reply:
We discussed examples mainly and where will those be located - to summarize, the conclusion we agreed was that first we will create and review examples, then we will update the documentation. 

This particular paragraph is mentioning the FPGA Addon Specialty IO, not the examples.  The FPGA Addon Specialty IO is one of the dependencies covered by the broader dependencies tracking project that we are discussing for some time. To my understanding, this particular toolkit will be installed with other dev toolkits, hopefully in vi.lib. As such, already during the ATS phase, the "Submodules" folder should disappear from the addon project structure as it will not be necessary anymore. 

Moreover, as it is at this point, after building the FPGA Addon CD, only addon specific files/libraries are moved to the VS CD Folder. No Specialty IO files are moved/copied to that location. As such, the path specified in the Quick Start Guide does NOT exist as of yet. This is the reason why I mentioned that the location should be discussed .... 

My opinion as of now - I think this comment should be removed from the Quick Start Guide. The toolkit should be installed in the LV vi.lib directory together with other dev dependencies. Instead, I think we should have there an example (LV Project + VS Project) that uses somehow the Specialty IO toolkit .... maybe an adaptation of a test case or one of the existing, original examples, imho.

Reply:
Makes sense, then please add a task under the "Examples" user story to update these locations.

Reply:
Sure thing ;)